ako-gateway-api/tests/gateway_test.go

```diff
@@ -0,0 +1,187 @@
+	os.Setenv("CLUSTER_NAME", "cluster")
+	os.Setenv("CLOUD_NAME", "CLOUD_VCENTER")
+	os.Setenv("SEG_NAME", "Default-Group")
+	os.Setenv("POD_NAMESPACE", utils.AKO_DEFAULT_NS)
+
+	akoControlConfig := akogatewayapilib.AKOControlConfig()
+	akoControlConfig.SetEventRecorder(lib.AKOGatewayEventComponent, kubeClient, true)
+	registeredInformers := []string{
+		utils.ServiceInformer,
+		utils.EndpointInformer,
+		utils.SecretInformer,
+		utils.NSInformer,
+		utils.ConfigMapInformer,
+	}
+	args := make(map[string]interface{})
+	utils.NewInformers(utils.KubeClientIntf{ClientSet: kubeClient}, registeredInformers, args)
+	integrationtest.InitializeFakeAKOAPIServer()
+
+	defer integrationtest.AviFakeClientInstance.Close()
+	ctrl = gwk8s.SharedGatewayController()
+	ctrl.InitGatewayAPIInformers(gwClient)
+
+	stopCh := utils.SetupSignalHandler()
+	ctrl.Start(stopCh)
+	keyChan = make(chan string)
+
+	ctrl.DisableSync = false
+
+	ctrl.SetupEventHandlers(k8s.K8sinformers{Cs: kubeClient})
+	numWorkers := uint32(1)
+	ctrl.SetupGatewayApiEventHandlers(numWorkers)
+	setupQueue(stopCh)
+	os.Exit(m.Run())
+}
+
+func TestGatewayCUD(t *testing.T) {
+	gateway := gatewayv1beta1.Gateway{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "gateway.networking.k8s.io/v1beta1",
+			Kind:       "Gateway",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "gw-example",
+			Namespace: "default",
+		},
+		Spec:   gatewayv1beta1.GatewaySpec{},
+		Status: gatewayv1beta1.GatewayStatus{},
+	}
+	SetGatewayGatewayClass(&gateway, "gw-class-example")
+	AddGatewayListener(&gateway, "listener-example", 80, gatewayv1beta1.HTTPProtocolType, false)
+	SetListenerHostname(&gateway.Spec.Listeners[0], "foo.example.com")
+
+	//create
+	gw, err := gwClient.GatewayV1beta1().Gateways("default").Create(context.TODO(), &gateway, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Couldn't create gateway, err: %+v", err)
+	}
+	t.Logf("Created gateway %+v", gw.Name)
+	waitAndverify(t, "Gateway/default/gw-example")
+
+	//update
+	SetGatewayGatewayClass(&gateway, "gw-class-new")
+	gw, err = gwClient.GatewayV1beta1().Gateways("default").Update(context.TODO(), &gateway, metav1.UpdateOptions{})
+	if err != nil {
+		t.Fatalf("Couldn't update gateway, err: %+v", err)
+	}
+	t.Logf("Updated gateway %+v", gw.Name)
+	waitAndverify(t, "Gateway/default/gw-example")
+
+	//delete
+	err = gwClient.GatewayV1beta1().Gateways("default").Delete(context.TODO(), gateway.Name, metav1.DeleteOptions{})
+	if err != nil {
+		t.Fatalf("Couldn't update gateway, err: %+v", err)
+	}
+	t.Logf("Deleted gateway %+v", gw.Name)
+	waitAndverify(t, "Gateway/default/gw-example")
+}
+
+func TestGatewaClassyCUD(t *testing.T) {
+	gatewayClass := gatewayv1beta1.GatewayClass{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "gateway.networking.k8s.io/v1beta1",
+			Kind:       "GatewayClass",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "gw-class-example",
+		},
+		Spec: gatewayv1beta1.GatewayClassSpec{
+			ControllerName: "ako.vmware.com/avi-lb",
```

Comment:
Are we having a validation logic to ignore the gatewayclass if the controllerName is not given as `"ako.vmware.com/avi-lb"`? Shall we add a UT for that in the next PR?